pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -246,7 +246,29 @@ rec {
           enableParallelChecking = attrs.enableParallelChecking or true;
         } // lib.optionalAttrs (hardeningDisable != [] || hardeningEnable != []) {
           NIX_HARDENING_ENABLE = enabledHardeningOptions;
-        } // lib.optionalAttrs (stdenv.buildPlatform.isDarwin) {
+        } // lib.optionalAttrs (stdenv.hostPlatform.isx86_64 && stdenv.hostPlatform.platform ? gcc.arch) ({
+          requiredSystemFeatures = attrs.requiredSystemFeatures or [] ++ [ "gccarch-${stdenv.hostPlatform.platform.gcc.arch}" ];
+          NIX_ENFORCE_NO_NATIVE = true;
+        } // lib.optionalAttrs ( !(attrs ? NIX_GCC_ARCH_FLAGS) # some packages might need customisation, give them priority
+                              && ( # older compilers (for example bootstrap's GCC5) fail with -march=too-modern-cpu
```

Comment:
That failure might be good to have. Otherwise it would be unclear why march isn’t getting set. I would consider it a legitimate error

Reply:
It makes would impossible bootstrapping a closure with ```gcc.arch=skylake``` because it uses GCC 5
Bootstrapping stages should not set ```platform.gcc.arch``` to work around it.
Upgrading GCC might be a simpler solution :)

Reply:
Ugh... yeah we have gcc5 in bootstrap tools but gcc7 in unstable